analyses/mwi/dryspells_trigger/exploration/2021_2022_postseason_data_analysis.R

```diff
@@ -0,0 +1,148 @@
+               rainy_bin = ifelse(rainfall >= 3, 1, 0)) %>% # binary: was >= 3mm of rain received? Rainy day definition from February 2022 bulletin from DCCMS
+      select(cell, date_chr, dates, month, week, rainfall, rainy_bin) %>%
+      arrange(cell, dates) 
+
+#####
+# analysis
+#####
+
+## 
+# Rainy season stats
+##
+
+# compute rolling sums
+stats <- lf %>%
+  group_by(cell) %>%
+  arrange(cell, dates) %>%
+  dplyr::mutate(roll_sum_next_10d = ifelse(!is.na(rainfall), zoo::rollsum(rainfall, k = 10, fill = NA, align = 'left'), NA), # sum of precipitation in 10-day period starting today
+         start_10d_ds_bin = ifelse(roll_sum_next_10d < 2, 1, 0), # is this the first day of a dry 10-day period?
+         roll_sum_foll_15d = ifelse(!is.na(rainfall), zoo::rollsum(lead(rainfall, 1), k = 15, fill = NA, align = 'left'), NA), # sum of precipitation in 15-day period starting tomorrow (lead=1)
+         foll_by_15d_ds_bin = ifelse(roll_sum_foll_15d <= 25, 1, 0) # is tomorrow the first day of a 15-d dry period?
+  )
+
+# Onsets:
+# First day of a period after 1 Nov with at least 40mm of rain over 10 days 
+# AND no 10 consecutive days with less than 2mm of total rain in the following 30 days (DCCMS 2008).
+onsets <- stats %>%
+             group_by(cell) %>%
+             arrange(dates) %>%
+             mutate(followed_by_ds = ifelse(!is.na(rainfall), zoo::rollsum(lead(start_10d_ds_bin, 1), k = 20, align = 'left') >= 1, NA)) %>% # boolean: is there at least one 10-day period with cum sum less than 2 in the next 20 days starting tomorrow (lead = 1)? k=20th day is the last chance to start a 10d-period within 30-day period
+             filter(dates >= '2021-11-01' & roll_sum_next_10d >= 40 & followed_by_ds == FALSE) %>%
+             slice(which.min(dates)) %>% # retrieve earliest date that meets criterion per cell 
+             ungroup() %>%
+             select(cell, date_chr, dates, roll_sum_next_10d, followed_by_ds) %>%
+             rename(onset_date_chr = date_chr, onset_date = dates) %>%
+             mutate(onset_days_since_1nov = as.numeric(difftime(onset_date, as.Date("2021-11-01"), unit = "days"))) %>% # Day 0 = 1 Nov
+             as.data.frame()
+
+# Cessations:
+# Last day before a 15-day period after 15 March with 25mm or less of rain
+cessations <- stats %>%
+            group_by(cell) %>%
+            arrange(dates) %>%
+            filter(dates >= '2022-03-15' & foll_by_15d_ds_bin == TRUE) %>%  # on or after 15 March
+            slice(which.min(dates)) %>% # retrieve earliest date that meets criterion per cell 
+            ungroup() %>%
+            select(cell, date_chr, dates, roll_sum_foll_15d, foll_by_15d_ds_bin) %>%
+            rename(cessation_date_chr = date_chr, cessation_date = dates) %>%
+            mutate(cessation_days_since_1nov = as.numeric(difftime(cessation_date, as.Date("2021-11-01"), unit = "days"))) %>% # Day 0 = 1 Nov
+            as.data.frame()
```

Comment:
I guess because `tibble` is producing an error somewhere else?

Reply:
mostly because printing more than 5 rows from a tibble is a pain. :) I can remove that, it's more of a personal preference for debugging.